apps/state/lib/state/schedule.ex

```diff
@@ -141,8 +141,25 @@ defmodule State.Schedule do
     {:ok, overridden_state, timeout_or_hibernate}
   end
 
-  # Converts routes and stops into workable ids
   @spec convert_filters(filter_opts) :: convert_filters
+  defp convert_filters(%{routes: routes, stops: stops} = filters) do
+    route_ids =
+      routes
+      |> Stream.uniq()
+      |> MapSet.new()
```

Comment:
`MapSet.new` will already ensure that the values are unique: you don't need the extra check.